seahub/profile/models.py

```diff
@@ -162,3 +164,9 @@ def update_profile_cache(sender, instance, **kwargs):
     Set profile data to cache when profile data change.
     """
     refresh_cache(instance.user)
+
+@receiver(institution_deleted)
+def remove_user_for_inst_deleted(sender, **kwargs):
```

Comment:
Should only set profile.institution to empty, instead of delete the whole profile